.evergreen/run-tox.sh

```diff
@@ -39,7 +39,7 @@ if [ "${INSTALL_TOX}" = "true" ]; then
         source motorenv/Scripts/activate
     fi
     set -o xtrace
-    pip install tox>=3.18
+    pip install "tox>=3.18,<4"
```

Comment:
Why make this change?

Reply:
```
[2023/02/14 17:07:14.554] tornado6-py38: failed with env name tornado6-py38 conflicting with base python c:\python\Python38\python.exe
 [2023/02/14 17:07:14.561]   tornado6-py38: FAIL code 1 (0.02 seconds)
```

Reply:
Ah, does tox 4.0+ not support Python 3.8?

Reply:
Let me try removing it again, tox 4 came out in December, we most likely would have seen this before now.

Reply:
Ah, it appears to be a case sensitivity issue on Windows: `Python3.8` ~= `python 3.8` in our `basepython` config.

Reply:
I believe this was caused by https://github.com/tox-dev/tox/pull/2824, which was released on 5 Jan and first started affecting our waterfall on 11 Jan.